src/main/java/edu/greenblitz/tobyDetermined/subsystems/DriveTrain/IO/SimModuleIO.java

```diff
@@ -0,0 +1,103 @@
+package edu.greenblitz.tobyDetermined.subsystems.DriveTrain.IO;
+
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.tobyDetermined.subsystems.swerve.SwerveChassis;
+import edu.greenblitz.utils.Conversions;
+import edu.greenblitz.utils.motors.GBFalcon;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import org.littletonrobotics.junction.Logger;
+
+public class SimModuleIO implements IModuleIO {
+
+    private final DCMotorSim linearMotor;
+    private final DCMotorSim angularMotor;
+    private IModuleIOInputsAutoLogged lastInputs = new IModuleIOInputsAutoLogged();
+    private final SwerveChassis.Module module;
+    private final PIDController angularController = new PIDController(
+            RobotMap.Swerve.SimulationSwerve.angularController.kP,
+            RobotMap.Swerve.SimulationSwerve.angularController.kI,
+            RobotMap.Swerve.SimulationSwerve.angularController.kD
+    );
+
+    private double angularAppliedVoltage, linearAppliedVoltage;
+
+    public SimModuleIO(SwerveChassis.Module module) {
+        this.module = module;
+        System.out.println("[Init] simulation swerve module " + this.module.name());
+
+        this.linearMotor = new DCMotorSim(
+                DCMotor.getFalcon500(1),
+                RobotMap.Swerve.SdsSwerve.LIN_GEAR_RATIO,
+                0.01
+                );
+        this.angularMotor = new DCMotorSim(
+                DCMotor.getFalcon500(1),
+                RobotMap.Swerve.SdsSwerve.ANG_GEAR_RATIO,
+                0.0001
+
+        );
+    }
+
+
+    @Override
+    public void setLinearVelocity(double speed) {
+        final double power = speed / RobotMap.Swerve.MAX_VELOCITY;
+        final double voltage = power * 12;
```

Comment:
What is 12?

Reply:
its the amount of milliseconds it took you to think about this.